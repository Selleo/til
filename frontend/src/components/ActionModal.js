import { useEffect } from 'react'
import Modal from 'react-modal'

const ActionModal = props => {
  const { message, action, isOpen, setIsOpen } = props

  const customStyles = {
    content: {
      top: '50%',
      left: '50%',
      right: 'auto',
      bottom: 'auto',
      transform: 'translate(-50%, -50%)',
      backgroundColor: '#272B2F',
      border: 'none',
      maxWidth: '1200px',
      width: '90%',
    },
  }

  useEffect(() => {
    if (process.env.NODE_ENV !== 'test') {
      Modal.setAppElement('#root')
    }
  }, [])

  const handleCancel = () => {
    setIsOpen(!isOpen)
  }

  const handleLeave = () => {
    action()
    setIsOpen(!isOpen)
  }

  return (
    <Modal
      isOpen={isOpen}
      style={customStyles}
      shouldCloseOnOverlayClick
      shouldCloseOnEsc
      overlayClassName="Overlay action-modal"
      appElement={document.getElementById('root')}
    >
      <div className="action-modal__content">
        <h3 className="action-modal__header">{message}</h3>
        <div className="action-modal__buttons">
          <button className="cancel-button" onClick={handleCancel}>
            Stay
          </button>
          <button className="delete-post-btn" onClick={handleLeave}>
            Leave anyway
          </button>
        </div>
      </div>
    </Modal>
  )
}

export default ActionModal
