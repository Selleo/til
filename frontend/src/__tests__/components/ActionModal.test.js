import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import ActionModal from '../../components/ActionModal'

describe('ActionModal', () => {
  it('should render', () => {
    const { baseElement } = render(
      <ActionModal isOpen={true} message={'test'} />
    )
    expect(baseElement).toMatchSnapshot()
    expect(screen.getByText('Stay')).toBeInTheDocument()
    expect(screen.getByText('Leave anyway')).toBeInTheDocument()
  })
  it('should call setIsOpen prop on "stay" button click', () => {
    const setIsOpenMock = jest.fn()
    render(<ActionModal setIsOpen={setIsOpenMock} isOpen message={'test'} />)
    fireEvent.click(screen.getByText('Stay'))

    expect(setIsOpenMock).toHaveBeenCalled()
  })
  it('should call setIsOpen and action prop on "leave" button click', () => {
    const setIsOpenMock = jest.fn()
    const actionMock = jest.fn()
    render(
      <ActionModal
        setIsOpen={setIsOpenMock}
        action={actionMock}
        isOpen
        message={'test'}
      />
    )
    fireEvent.click(screen.getByText('Leave anyway'))

    expect(actionMock).toHaveBeenCalled()
  })
})
