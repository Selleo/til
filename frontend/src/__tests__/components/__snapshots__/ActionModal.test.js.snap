// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionModal should render 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay Overlay action-modal"
    >
      <div
        aria-modal="true"
        class="ReactModal__Content"
        role="dialog"
        style="position: absolute; top: 50%; left: 50%; background: rgb(255, 255, 255); overflow: auto; border-radius: 4px; outline: none; padding: 20px; transform: translate(-50%, -50%); background-color: rgb(39, 43, 47); max-width: 1200px; width: 90%;"
        tabindex="-1"
      >
        <div
          class="action-modal__content"
        >
          <h3
            class="action-modal__header"
          >
            test
          </h3>
          <div
            class="action-modal__buttons"
          >
            <button
              class="cancel-button"
            >
              Stay
            </button>
            <button
              class="delete-post-btn"
            >
              Leave anyway
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
